nixos/modules/services/misc/omnom.nix

```diff
@@ -0,0 +1,178 @@
+                default = "${cfg.dataDir}/db.sqlite3";
+                description = "Database connection URI.";
+                defaultText = lib.literalExpression ''
+                  "''${config.services.omnom.dataDir}/db.sqlite3"
+                '';
+              };
+              type = lib.mkOption {
+                type = lib.types.str;
+                default = "sqlite";
+                description = "Database type.";
+              };
+            };
+            server = {
+              address = lib.mkOption {
+                type = lib.types.str;
+                default = "127.0.0.1:7331";
+                description = "Server address.";
+              };
+              secure_cookie = lib.mkOption {
+                type = lib.types.bool;
+                default = true;
+                description = "Limit cookies to secure channel.";
+              };
+            };
+            storage = {
+              root = lib.mkOption {
+                type = lib.types.str;
+                default = "${cfg.dataDir}/static/data";
+                defaultText = lib.literalExpression ''
+                  "''${config.services.omnom.dataDir}/static/data"
+                '';
+                description = "Where the snapshots are saved.";
+              };
+            };
+          };
+        };
+        default = { };
+      };
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    environment.etc."omnom/config.yml".source = pkgs.writers.writeYAML "omnom-config.yml" cfg.settings;
+
+    services.omnom.settings = {
+      app.debug = lib.mkDefault cfg.debug;
+      storage.type = lib.mkDefault "fs";
+    };
+
+    systemd.services.omnom = {
+      path = [ pkgs.rsync ];
+
+      serviceConfig = {
+        DynamicUser = isDynamicUser;
+        User = cfg.user;
+        Group = cfg.group;
+        ExecStart =
+          let
+            args = lib.cli.toGNUCommandLineShell { } {
+              config = "/etc/omnom/config.yml";
+              d = cfg.debug;
+            };
+          in
+          "${lib.getExe cfg.package} listen ${args}";
+        StateDirectory = lib.mkIf (cfg.dataDir == "/var/lib/omnom") "omnom";
+        StateDirectoryMode = lib.mkIf (cfg.dataDir == "/var/lib/omnom") "0750";
+        WorkingDirectory = cfg.dataDir;
+        Restart = "on-failure";
+        RestartSec = "10s";
+      };
+      after = [
+        "network.target"
+        "systemd-tmpfiles-setup.service"
+      ];
+      wantedBy = [ "multi-user.target" ];
+
+      # These need to be present in the working directory
+      preStart = ''
+        for dir in templates; do
+          if [ ! -e "${cfg.dataDir}"/$dir ]; then
+            rsync -a --chmod=u=rwX,go=rX "${cfg.package}/share/$dir" "${cfg.dataDir}/"
+          fi
+        done
+      '';
+    };
+
+    # If the service user is not dynamic, a normal group must exist
+    users.groups = lib.mkIf ((!isDynamicUser) && (cfg.group == "omnom")) { omnom = { }; };
```

Comment:
> Currently, you can't register users from the command line if the service is running as a DynamicUser. This is because the Omnom CLI tool runs as the user that starts it and it requires that all the files it needs are in its CWD.

In that case we should create a wrapper script that changes the user and set the CWD to the dataDir. See the `occ` wrapper in nextcloud module for example.